src/components/Header/UserPreferences/Automation/index.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import { useAutomation } from '../../../../@context/Automation/AutomationProvider'
+import Tooltip from '../../../@shared/atoms/Tooltip'
+import Details from './Details'
+import Transaction from '@images/transaction.svg'
+import Caret from '@images/caret.svg'
+import Lock from '@images/lock.svg'
+import classNames from 'classnames/bind'
+import stylesIndex from '../index.module.css'
+import styles from './index.module.css'
+
+const cx = classNames.bind(styles)
+
+export default function Automation(): ReactElement {
+  const {
+    autoWallet,
+    isAutomationEnabled,
+    hasValidEncryptedWallet,
+    balance,
+    nativeBalance
+  } = useAutomation()
+
+  const [hasError, setHasError] = useState<boolean>()
+  const [hasWarning, setHasWarning] = useState<boolean>(false)
+
+  useEffect(() => {
+    balance &&
+      setHasWarning(
```

Comment:
suggestion: The `hasWarning` state is set based on the balance of individual tokens being less than or equal to 0. It might be more intuitive to rename this state to something like `hasZeroTokenBalance` for clarity.